### PR TITLE
[gmp] arm64 windows build support

### PR DESCRIPTION
**Pull request by eukarpov**
**Describe the pull request**

- #### What does your PR fix?
  PR adds arm64-windows support for gmp package by disabling assembly optimization 
  arm64 build passed all tests provided by gmp package. 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -698,6 +698,9 @@ nanodbc:x64-uwp=fail
 nanodbc:x64-linux=skip
 nativefiledialog:arm-uwp=fail
 nativefiledialog:x64-uwp=fail
+nettle:arm-uwp=fail
+nettle:arm64-windows=fail
+nettle:x64-osx=fail
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:703]: I see for arm-uwp and arm64-windows they were blocked by the dependency before:

https://github.com/microsoft/vcpkg/blob/5f83f03a1c7578e6ae04987916b34e018fb63964/ports/nettle/vcpkg.json#L8

However, I don't think that explains x64-osx, which was allowed by the previous `supports` expression is explained here. Did you see that actual failure?

Comment (eukarpov) [reply scripts/ci.baseline.txt:703]: yes, before applying changes in CI baseline
https://github.com/microsoft/vcpkg/runs/7792399616

Comment (BillyONeal) [reply scripts/ci.baseline.txt:703]:
```
In file included from ./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt1.c:26:
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.h:169:19: warning: this function declaration is not a prototype [-Wstrict-prototypes]
extern int getopt ();
                  ^
                   void
1 warning generated.
In file included from ./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.c:98:
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.h:169:19: warning: this function declaration is not a prototype [-Wstrict-prototypes]
extern int getopt ();
                  ^
                   void
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.c:146:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
extern char *getenv ();
                    ^
                     void
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.c:272:25: warning: unused parameter 'argc' [-Wunused-parameter]
_getopt_initialize (int argc, char *const *argv, const char *optstring,
                        ^
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.c:272:44: warning: unused parameter 'argv' [-Wunused-parameter]
_getopt_initialize (int argc, char *const *argv, const char *optstring,
                                           ^
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.c:1188:1: warning: no previous prototype for function 'getopt' [-Wmissing-prototypes]
getopt (int argc, char *const *argv, const char *optstring)
^
./../src/1f35bea1c1-b3c5d0ac7b.clean/getopt.h:169:12: note: this declaration is not a prototype; add parameter declarations to make it one
extern int getopt ();
           ^
5 warnings generated.
ecc-mod-arith.o.d:55: *** missing separator.  Stop.
make[1]: *** [eccdata.stamp] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

Is it possible that the missing separator is coming from something controlled by this enable assembly option?

I'm not sure a `ci.baseline.txt` change is the right one for this...

**Review by BillyONeal — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thanks!

**Pull request merged**